src/server/dao/cc-proxy/SpineTreeDao.ts

```diff
@@ -0,0 +1,61 @@
+import { BaseDao } from '@hmh/nodejs-base-server';
+import { SpineTree as Model } from '../../model/cc-proxy/SpineTree';
+import { Spine as Source } from '../../model/cc/Spine';
+import { SpineDao as ModelDAO } from '../cc/SpineDao';
+
+export class SpineTreeDao extends BaseDao<Model> {
+    // Factory method
+    public static getInstance(): SpineTreeDao {
+        if (!SpineTreeDao.instance) {
+            SpineTreeDao.instance = new SpineTreeDao();
+        }
+        return SpineTreeDao.instance;
+    }
+
+    private static instance: SpineTreeDao;
+    private spineDao: ModelDAO;
+    private spines: Map<string, Model>;
```

Comment:
This DAO will be mostly used to get the list of Spine elements.
I think keeping a cache of the list makes more sense than transforming the list in a map (once) and then to list (many times)...
Please, change the attribute type from a `Map` to an `Array`.

This will provide the additional benefits to get rid off `populateSpines()` because `getSpines()` will be able to do everything, and to have only one function dealing with the cache reference...

``` javascript
private getSpines(): Promise<Model[]> {
    if (this.spines.length === 0) {
        this.spines = await this.spineDao.query();
    }
    return this.spines;
}

private getSpine(id: string): Promise<Model> {
    let spine: Spine = await this.getSpines().find((element: Spine): boolean => element.id === id);
    if (spine === undefined) {
        throw new NotFoundException(`SpineTree with id '${id}' not found`);
    }
    return spine;
}
```